### PR TITLE
Support kind blacklist in pruned deployments

### PR DESCRIPTION
**Pull request by clarketm**
`core/v1/Binding` is not a prunable resource, thus adding a kind blacklist to exclude it.

fix: [deploy-prow_test-infra_postsubmit](https://storage.googleapis.com/istio-prow/logs/deploy-prow_test-infra_postsubmit/47/build-log.txt)
blocking: #2525

### CODE REVIEW

File: prow/Makefile.gcloud.mk [hunk 1]
```diff
@@ -41,11 +41,8 @@ configure-docker: activate-serviceaccount
 
 get%api-resources:
 	$(eval PRUNE_WL=$(shell KUBECONFIG=$(KUBECONFIG) \
```
Comment (fejta) [inline prow/Makefile.gcloud.mk:43]: There is WAY too much eval everywhere, which violates our style guidelines for bash.

http://google.github.io/styleguide/shellguide.html#s6.6-eval

We should also try to putting logic in makefiles as these are not portable across repos

Why is this necessary here?

Comment (clarketm) [reply prow/Makefile.gcloud.mk:43]: This is to _set_ to global `PRUNE_WL` `make` variable used in the other `make` deploy rules in `prow/Makefile`, so a single target can handle all of these. I understand this is difficult to follow. I can work on a revision but I would like to get this specific change to `api-resources.py` in to fix the deploy job.

Comment (clarketm) [reply prow/Makefile.gcloud.mk:43]: I agree. I think a better approach is to pull this out of the Makefile like you suggest, into a script or program.

Comment (clarketm) [reply prow/Makefile.gcloud.mk:43]: I created this: https://github.com/istio/test-infra/issues/2552.  Will handle as a follow-up.

**Review by fejta — APPROVED** [review]
